Client/MirObjects/PlayerObject.cs

```diff
@@ -531,13 +1054,22 @@ public virtual void SetLibraries()
             }
 
             //Fishing
-            if (HasFishingRod)
+           if (HasFishingRod)
             {
                 if (CurrentAction == MirAction.FishingCast || CurrentAction == MirAction.FishingWait || CurrentAction == MirAction.FishingReel)
                 {
-                    WeaponLibrary1 = 0 < Libraries.Fishing.Length ? Libraries.Fishing[Weapon - 49] : null;
-                    WeaponLibrary2 = null;
-                    WeaponOffSet = -632;
+                    if (Class < MirClass.HighWarrior)//stupple
+                    {
+                        WeaponLibrary1 = 0 < Libraries.Fishing.Length ? Libraries.Fishing[Weapon - 49] : null;
+                        WeaponLibrary2 = null;
+                        WeaponOffSet = -632;
+                    }
+                    else
+                    {
+                        WeaponLibrary1 = 0 < (Gender == MirGender.Male ? Libraries.UpFishingM.Length : Libraries.UpFishingF.Length) ? (Gender == MirGender.Male ? Libraries.UpFishingM[Weapon - 49] : Libraries.UpFishingF[Weapon - 49]) : null;
+                        WeaponLibrary2 = null;
+                        WeaponOffSet = -600;
```

Comment:
you sure this offset is correct? the original one was -632 however now that only applies to highWarrior